core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,12 +54,33 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).then(function() {
-          // TODO(bhenning): Consolidate the backend rights domain objects and
-          // implement a frontend activity rights domain object. The rights
-          // being passed in here should be used to create one of those objects.
-          if (successCallback) {
-            successCallback();
+        $http.put(collectionRightsUrl, putParams).then(function(response) {
+          // Check if the response from the backend does not contradict
+          // putParams.
+          var isPrivateResponse = (response.data.status === 'private');
+          if (isPrivateResponse === isPublic) {
+            if (isPublic) {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'published, but response.data.is_private returned true.');
+            } else {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'unpublished, but response.data.is_private returned false.');
+            }
+            if (errorCallback) {
+              errorCallback(response.data);
+            }
+          } else {
+            // An additional fetch is required because the GET response object
```

Comment:
I modified the response from `PUT` so that it had at least what was the response from `GET`, but Sean suggested against manipulating the `PUT` response. I wouldn't want to return what was in the `PUT` response in the `GET` either since it doesn't match what we want here.